packages/flame/lib/src/game/base_game.dart

```diff
@@ -238,6 +238,69 @@ class BaseGame extends Game with FPSCounter {
   /// show extra information on the screen when debug mode is activated
   bool debugMode = false;
 
+  /// Changes the priority of [component] and reorders the games component list.
+  ///
+  /// Returns true if changing the component's priority modified one of the
+  /// components that existed directly on the game and false if it
+  /// either was a child of another component, if it didn't exist at all or if
+  /// it was a component added directly on the game but its priority didn't
+  /// change.
+  bool changePriority(
+    Component component,
+    int priority, {
+    bool reorderRoot = true,
+  }) {
+    if (component.priority == priority) {
+      return false;
+    }
+    if (components.contains(component)) {
+      component.changePriorityWithoutResorting(priority);
+      if (reorderRoot) {
+        components.rebalanceAll();
+      }
+      return true;
+    } else if (component is BaseComponent) {
+      component.changePriorityWithoutResorting(priority);
+      BaseComponent? foundParent;
+      for (final parent in components.whereType<BaseComponent>()) {
+        if (parent == component.parent) {
+          foundParent = parent;
+          break;
+        }
+        parent.propagateToChildren<BaseComponent>((child) {
+          if (child == component.parent) {
+            foundParent = child;
+            return false;
+          }
+          return true;
+        });
+        if (foundParent != null) {
+          break;
+        }
+      }
+      foundParent?.reorderChildren();
```

Comment:
technically this will rebalance the same component multiple times when using `changePriorities` with siblings, not sure if that is a huge issue.

Reply:
Fixed this issue by putting all the parent in a set first and then reordering their children